test/framework/releaseVersions.go

```diff
@@ -28,10 +37,15 @@ func (e *ClusterE2ETest) GetLatestMinorReleaseFromMain() *releasev1alpha1.EksARe
 		e.T.Fatalf("Releases manifest doesn't contain latest release %s", releases.Spec.LatestVersion)
 	}
 
-	return latestRelease
+	binaryPath, err := e.getBinary(latestRelease)
+	if err != nil {
+		e.T.Fatalf("Failed getting binary for latest release: %s", err)
+	}
+
+	return binaryPath
 }
 
-func (e *ClusterE2ETest) GetLatestMinorReleaseFromReleaseBranch(releaseBranchVersion *semver.Version) *releasev1alpha1.EksARelease {
+func (e *ClusterE2ETest) GetLatestMinorReleaseFromReleaseBranch(releaseBranchVersion *semver.Version) string {
```

Comment:
Good point. I can change it to `GetLatestMinorReleaseBinaryFromReleaseBranch` because it'll only download the binary if it doesn't exist, but it'll always "get" the binary to use in the E2E test.